corehq/apps/data_interfaces/forms.py

```diff
@@ -94,3 +98,179 @@ def __init__(self, *args, **kwargs):
                 type="submit"
             )
         )
+
+
+class AddAutomaticCaseUpdateRuleForm(forms.Form):
+    name = TrimmedCharField(
+        label=ugettext_lazy("Name"),
+        required=True,
+    )
+    case_type = TrimmedCharField(
+        label=ugettext_lazy("Case Type"),
+        required=True,
+    )
+    server_modified_boundary = forms.IntegerField(
+        required=True,
+    )
+    conditions = forms.CharField(
+        required=True,
+    )
+    close_case = forms.BooleanField(
+        label=ugettext_lazy("Close the Case"),
+        required=False,
+    )
+    update_case = forms.BooleanField(
+        label=ugettext_lazy("Set a case property"),
+        required=False,
+    )
+    update_property_name = TrimmedCharField(
+        label=ugettext_lazy("Property"),
+        required=False,
+    )
+    update_property_value = TrimmedCharField(
+        label=ugettext_lazy("Value"),
+        required=False,
+    )
+
+    @property
+    def current_values(self):
+        values = {}
+        for field_name in self.fields.keys():
+            value = self[field_name].value()
+            if field_name == 'conditions':
+                value = value or '[]'
+                value = json.loads(value)
+            values[field_name] = value
+        return values
+
+    def __init__(self, *args, **kwargs):
+        super(AddAutomaticCaseUpdateRuleForm, self).__init__(*args, **kwargs)
+        self.helper = FormHelper()
+        self.helper.form_class = 'form form-horizontal'
+        self.helper.label_class = 'col-sm-1 col-md-1'
+        self.helper.field_class = 'col-sm-3 col-md-2'
+        self.helper.form_method = 'POST'
+        self.helper.form_action = '#'
+        self.helper.layout = Layout(
+            Fieldset(
+                _("Basic Information"),
+                Field(
+                    'name',
+                    **{'ng-model': 'name'}
+                ),
+                Field(
+                    'case_type',
+                    **{'ng-model': 'case_type'}
+                ),
+            ),
+            Fieldset(
+                _("Conditions"),
+                Field(
+                    'conditions',
+                    type='hidden',
+                    **{'ng-value': 'conditions'}
+                ),
+                Div(**{'ng-include': '', 'src': "'conditions.tpl'"}),
+            ),
+            Fieldset(
+                _("Action"),
+                Field(
+                    'close_case',
+                    disabled=True,
+                    checked='',
+                ),
+                Field(
+                    'update_case',
+                    **{'ng-model': 'update_case'}
+                ),
+                Div(
+                    Field(
+                        'update_property_name',
+                        **{'ng-model': 'update_property_name'}
```

Comment:
`ng_model='update_property_name'` also works here. all underscores in `Field` for non-reserved kwargs turn into dashes. @dimagi/team-commcare-hq also good for other devs to note